docs/designs/preprocess_with_analysis.md

```diff
@@ -0,0 +1,165 @@
+# Preprocess with Analysis Result Design
+This document describes the design of preprocessing feature inputs using analysis results.
+
+## Motivation
+Before preprocessing the feature inputs, we need to analyze the training dataset to collect the feature statistics. For example, we need the mean and standard deviation to normalize a numeric value, `vocabulary` to lookup a string value to an integer id and `boundary` to discretize a numeric value. Using SQLFlow, the training dataset is usually a table saved in MySQL or MaxCompute and other databases, so we can use SQL to analyze the training table. During [data transformation pipeline](https://github.com/sql-machine-learning/elasticdl/blob/develop/docs/designs/data_transform.md), we may launch a pod to analyze the training table and then submit the ElasticDL training job. So, the design is to solve how to pass the analysis results into the pods of an ElasticDL training job.
+
```

Comment:
There are two steps in preprocessing: analysis and transformation. The results generated by analysis will be used in transformation.  It seems that it's a workflow.  Please refer to [here](https://github.com/sql-machine-learning/fluid/blob/develop/doc/design.md).

I am not sure if we could use workflow to describe the data processing pipeline, and also the dependencies between steps.

Reply:
Yes, it is a pipeline. We will use workflow to describe the data preprocessing and training job when we use it in SQLFlow. The design only discusses how to pass the results generated by analysis into training pods to build the model. We can orchestrate the analysis containers and training containers using k8s workflow or PAI-GUI.